### ISSUE TITLE
get_task failing

### ISSUE THREAD

**Issue by ArlindKadra** [openml/openml-python#538]
#### Description
Getting a certain clustering task is failing. Task shows fine on the server.
#### Steps/Code to Reproduce

```
import openml

result = openml.tasks.functions.get_task(126033)
print(result)

```
#### Expected Results
Actual task.
#### Actual Results
`FileNotFoundError: [Errno 2] No such file or directory:`

**Comment by ArlindKadra** [comment]
input is not a list but a dictionary.
https://github.com/openml/openml-python/blob/34c06df237dc3f953b0be77acedfc1f28b82abd5/openml/tasks/functions.py#L351

**Comment by ArlindKadra** [comment]
@janvanrijn  @mfeurer I am trying to fix the issue above and I have a question regarding the current implementation of tasks.
https://github.com/openml/openml-python/blob/811f9cee3bdbca535eb898282edab9ab6ee71594/openml/tasks/task.py#L10-L20

The base class upon which all of the tasks are built, has estimation procedure as a mandatory field, with type and parameters. The above issue is happening because the tasks has only 1 input. 

Are clustering tasks supposed to have a mandatory estimation procedure field and the above an outlier which should be removed or I should change the implementation because like this.

The easiest fix for the second option, would be to have the common args like that for all the other tasks and to pass none for the estimation procedure for the clustering task.

**Comment by mfeurer** [comment]
I think removing the estimation procedure from the base class and having it in the `OpenMLSupervisedTask` would be a clean solution.

**Comment by mfeurer** [comment]
Fixed by #541.
